File: Ejercicios Propuestos 1/Ejercicio 6/EJERCICIO_6.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    double angulo, seno_acumulado, termino, signo, factorial;
    int exponente;

    cout << "Ingrese el ángulo en radianes (entre -3.1416 y 3.1416):" << endl;
    cin >> angulo;

    if (angulo < -3.1416 || angulo > 3.1416) {
        cout << "Error: Ángulo fuera del rango permitido." << endl;
    } else {
        seno_acumulado = angulo;
        termino = angulo;
        signo = 1;
        factorial = 1;

        for (int i = 1; i <= 4; i++) {
            signo = signo * (-1);
            exponente = 2 * i + 1;

            factorial = factorial * (2 * i) * (2 * i + 1);
            termino = pow(angulo, exponente) / factorial;
            seno_acumulado = seno_acumulado + (signo * termino);
        }

        cout << "Seno aproximado de (" << angulo << ") = " << seno_acumulado << endl;
    }

    return 0;
}

File: Ejercicios Propuestos 1/Ejercicio 6/Ejercicio6.java
import java.util.Scanner;

public class Ejercicio6 {
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Ingrese el ángulo en grados (0 a 360): ");
        double angulo = scanner.nextDouble();
        
        if (angulo < 0 || angulo > 360) {
            System.out.println("Error: debe estar entre 0 y 360");
            return;
        }
    
        double radianes = Math.toRadians(angulo);
        
        double seno = radianes - (radianes * radianes * radianes) / 6 
                    + (radianes * radianes * radianes * radianes * radianes) / 120;
        
        System.out.println("Seno de " + angulo + "° = " + seno);
    }
}